File: components/api/api-modules-python/src/main/resources/META-INF/dirigible/python-modules/dirigible/io/zip.py
# Copyright (c) 2023 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
# All rights reserved. This program and the accompanying materials
# are made available under the terms of the Eclipse Public License v2.0
# which accompanies this distribution, and is available at
# http://www.eclipse.org/legal/epl-v20.html
# SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
# SPDX-License-Identifier: EPL-2.0

import bytes
import java


ZipFacade = java.type("org.eclipse.dirigible.components.api.io.ZipFacade")

def zip(sourcePath, zipTargetPath):
    ZipFacade.exportZip(sourcePath, zipTargetPath)

def unzip(zipPath, targetPath):
    ZipFacade.importZip(zipPath, targetPath)

def createZipInputStream(inputStream):
    native = ZipFacade.createZipInputStream(inputStream.native)
    return ZipInputStream(native)

class ZipInputStream:
    @staticmethod
    def __init__(self, native):
        self.native = native

    @staticmethod
    def getNextEntry(self):
        native = self.native.getNextEntry()
        zipEntry = ZipEntry(native)
        return zipEntry

    @staticmethod
    def read(self):
        native = ZipFacade.readNative(self.native)
        return bytes.toPythonBytes(native)

    @staticmethod
    def readNative(self):
        return ZipFacade.readNative(self.native)

    @staticmethod
    def readText(self):
        return ZipFacade.readText(self.native)

    @staticmethod
    def close(self):
        self.native.close()

def createZipOutputStream(outputStream):
    native = ZipFacade.createZipOutputStream(outputStream.native)
    return ZipOutputStream(native)

class ZipOutputStream:
    @staticmethod
    def __init__(self, native):
        self.native = native

    @staticmethod
    def createZipEntry(self, name):
        nativeNext = ZipFacade.createZipEntry(name)
        zipEntry = ZipEntry(nativeNext)
        self.native.putNextEntry(nativeNext)
        return zipEntry

    @staticmethod
    def write(self, data):
        native = bytes.toJavaBytes(data)
        ZipFacade.writeNative(self.native, native)

    @staticmethod
    def writeNative(self, data):
        ZipFacade.writeNative(self.native, data)

    @staticmethod
    def writeText(self, text):
        ZipFacade.writeText(self.native, text)

    @staticmethod
    def closeEntry(self):
        self.native.closeEntry()

    @staticmethod
    def close(self):
        self.native.finish()
        self.native.flush()
        self.native.close()

class ZipEntry:
    @staticmethod
    def __init__(self, native):
        self.native = native

    @staticmethod
    def getName(self):
        return self.native.getName()

    @staticmethod
    def getSize(self):
        return self.native.getSize()

    @staticmethod
    def getCompressedSize(self):
        return self.native.getCompressedSize()

    @staticmethod
    def getTime(self):
        return self.native.getTime()

    @staticmethod
    def getCrc(self):
        return self.native.getCrc()

    @staticmethod
    def getComment(self):
        return self.native.getComment()

    @staticmethod
    def isDirectory(self):
        return self.native.isDirectory()

    @staticmethod
    def isValid(self):
        return self.native is not None

File: components/api/api-modules-javascript/src/main/resources/META-INF/dirigible/modules/src/io/zip.ts
/*
 * Copyright (c) 2023 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 * SPDX-License-Identifier: EPL-2.0
 */
/**
 * API Zip
 */
import {bytes} from ".";
import { InputStream, OutputStream } from "./streams";
const ZipFacade = Java.type("org.eclipse.dirigible.components.api.io.ZipFacade");

export class Zip{

    public static zip(sourcePath: string, zipTargetPath: string): void {
        ZipFacade.exportZip(sourcePath, zipTargetPath);
    };

    public static unzip(zipPath: string, targetPath: string): void {
        ZipFacade.importZip(zipPath, targetPath);
    };

    public static createZipInputStream(inputStream: InputStream): ZipInputStream {
        const native = ZipFacade.createZipInputStream(inputStream.native);
        return new ZipInputStream(native);
    };
}

class ZipInputStream {

    constructor(private native: any){}

    public getNextEntry(): ZipEntry {
        const native = this.native.getNextEntry();
        const zipEntry = new ZipEntry(native);
        return zipEntry;
    };

    public read(): any {
        const native = ZipFacade.readNative(this.native);
        return bytes.toJavaScriptBytes(native);
    };

    public readNative(): any {
        return ZipFacade.readNative(this.native);
    };

    public readText(): string {
        return ZipFacade.readText(this.native);
    };

    public close(): void {
        this.native.close();
    };

};

export function createZipOutputStream(outputStream: OutputStream): ZipOutputStream {
    const native = ZipFacade.createZipOutputStream(outputStream.native);
    return new ZipOutputStream(native);
};

class ZipOutputStream {

   constructor(private native: any){}

    public createZipEntry(name: string): ZipEntry {
        const nativeNext = ZipFacade.createZipEntry(name);
        const zipEntry = new ZipEntry(nativeNext);
        this.native.putNextEntry(nativeNext);
        return zipEntry;
    };

    public write(data: any): void {
        const native = bytes.toJavaBytes(data);
        ZipFacade.writeNative(this.native, native);
    };

    public writeNative(data: any): void {
        ZipFacade.writeNative(this.native, data);
    };

    public writeText(text: string): void {
        ZipFacade.writeText(this.native, text);
    };

    public closeEntry(): void {
        this.native.closeEntry();
    };

    public close(): void {
        this.native.finish();
        this.native.flush();
        this.native.close();
    };

};

/**
 * ZipEntry object
 */
class ZipEntry {

    constructor(private native: any){}

    public getName(): string {
        return this.native.getName();
    };

    public getSize(): number {
        return this.native.getSize();
    };

    public getCompressedSize(): number {
        return this.native.getCompressedSize();
    };

    public getTime(): Date {
        return this.native.getTime();
    };

    public getCrc(): any {
        return this.native.getCrc();
    };

    public getComment(): Comment {
        return this.native.getComment();
    };

    public isDirectory(): boolean {
        return this.native.isDirectory();
    };

    public isValid(): boolean {
        return this.native !== null;
    };

}
